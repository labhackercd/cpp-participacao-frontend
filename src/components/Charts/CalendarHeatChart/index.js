import React from 'react';
//import Plot from 'react-plotly.js';
import Chart from "react-google-charts";


export default function CalendarHeatChart(props) {
  return (
      <Chart
        width={'1000px'}
        height={'350px'}
        chartType="Calendar"
        loader={<div>Carregando Gr√°fico</div>}
        data={props.data}
        options={props.options}
        rootProps={{ 'data-testid': '1' }}
    />
   );
}
