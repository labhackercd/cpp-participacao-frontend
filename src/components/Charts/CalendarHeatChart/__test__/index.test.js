import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from "enzyme";
import CalendarHeatChart from './../index';

const pieChartdata = {
    values:[
      [{ type: 'date', id: 'Date' }, { type: 'number', id: 'Won/Loss' }],
      [new Date(2012, 3, 13), 37032],
      [new Date(2012, 3, 14), 38024],
      [new Date(2012, 3, 15), 38024],
      [new Date(2012, 3, 16), 38108],
      [new Date(2012, 3, 17), 38229],
      [new Date(2013, 1, 4), 38177],
      [new Date(2013, 1, 5), 38705],
      [new Date(2013, 1, 12), 38210],
      [new Date(2013, 1, 13), 38029],
      [new Date(2013, 1, 19), 38823],
      [new Date(2013, 1, 23), 38345],
      [new Date(2013, 1, 24), 38436],
      [new Date(2013, 2, 10), 38447],
  ],
    options:{
    title: 'Red Sox Attendance',
  }
};

test("Snapshot should not have changes", () => {
    const component = mount(<CalendarHeatChart data={pieChartdata.values} options={pieChartdata.options}></CalendarHeatChart>);
    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});

it("Mounts the chart with the data", () => {
    const div = document.createElement("div")
    ReactDOM.render(<CalendarHeatChart data={pieChartdata.values} options={pieChartdata.options}></CalendarHeatChart>, div)
});
