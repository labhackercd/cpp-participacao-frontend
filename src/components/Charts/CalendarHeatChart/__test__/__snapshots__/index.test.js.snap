// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should not have changes 1`] = `
<CalendarHeatChart
  data={
    Array [
      Array [
        Object {
          "id": "Date",
          "type": "date",
        },
        Object {
          "id": "Won/Loss",
          "type": "number",
        },
      ],
      Array [
        2012-04-13T03:00:00.000Z,
        37032,
      ],
      Array [
        2012-04-14T03:00:00.000Z,
        38024,
      ],
      Array [
        2012-04-15T03:00:00.000Z,
        38024,
      ],
      Array [
        2012-04-16T03:00:00.000Z,
        38108,
      ],
      Array [
        2012-04-17T03:00:00.000Z,
        38229,
      ],
      Array [
        2013-02-04T02:00:00.000Z,
        38177,
      ],
      Array [
        2013-02-05T02:00:00.000Z,
        38705,
      ],
      Array [
        2013-02-12T02:00:00.000Z,
        38210,
      ],
      Array [
        2013-02-13T02:00:00.000Z,
        38029,
      ],
      Array [
        2013-02-19T03:00:00.000Z,
        38823,
      ],
      Array [
        2013-02-23T03:00:00.000Z,
        38345,
      ],
      Array [
        2013-02-24T03:00:00.000Z,
        38436,
      ],
      Array [
        2013-03-10T03:00:00.000Z,
        38447,
      ],
    ]
  }
  options={
    Object {
      "title": "Red Sox Attendance",
    }
  }
>
  <Chart
    chartActions={null}
    chartEvents={null}
    chartType="Calendar"
    chartWrapperParams={Object {}}
    className=""
    columns={null}
    controls={null}
    data={
      Array [
        Array [
          Object {
            "id": "Date",
            "type": "date",
          },
          Object {
            "id": "Won/Loss",
            "type": "number",
          },
        ],
        Array [
          2012-04-13T03:00:00.000Z,
          37032,
        ],
        Array [
          2012-04-14T03:00:00.000Z,
          38024,
        ],
        Array [
          2012-04-15T03:00:00.000Z,
          38024,
        ],
        Array [
          2012-04-16T03:00:00.000Z,
          38108,
        ],
        Array [
          2012-04-17T03:00:00.000Z,
          38229,
        ],
        Array [
          2013-02-04T02:00:00.000Z,
          38177,
        ],
        Array [
          2013-02-05T02:00:00.000Z,
          38705,
        ],
        Array [
          2013-02-12T02:00:00.000Z,
          38210,
        ],
        Array [
          2013-02-13T02:00:00.000Z,
          38029,
        ],
        Array [
          2013-02-19T03:00:00.000Z,
          38823,
        ],
        Array [
          2013-02-23T03:00:00.000Z,
          38345,
        ],
        Array [
          2013-02-24T03:00:00.000Z,
          38436,
        ],
        Array [
          2013-03-10T03:00:00.000Z,
          38447,
        ],
      ]
    }
    diffdata={null}
    formatters={null}
    getChartEditor={null}
    getChartWrapper={[Function]}
    graphID={null}
    graph_id={null}
    height="350px"
    legendToggle={false}
    legend_toggle={false}
    loader={
      <div>
        Carregando Gráfico
      </div>
    }
    options={
      Object {
        "title": "Red Sox Attendance",
      }
    }
    render={null}
    rootProps={
      Object {
        "data-testid": "1",
      }
    }
    rows={null}
    spreadSheetQueryParameters={
      Object {
        "gid": 1,
        "headers": 1,
      }
    }
    spreadSheetUrl={null}
    style={Object {}}
    toolbarID={null}
    toolbarItems={null}
    width="1000px"
  >
    <ContextProvider
      value={
        Object {
          "chartActions": null,
          "chartEvents": null,
          "chartType": "Calendar",
          "chartWrapperParams": Object {},
          "className": "",
          "columns": null,
          "controls": null,
          "data": Array [
            Array [
              Object {
                "id": "Date",
                "type": "date",
              },
              Object {
                "id": "Won/Loss",
                "type": "number",
              },
            ],
            Array [
              2012-04-13T03:00:00.000Z,
              37032,
            ],
            Array [
              2012-04-14T03:00:00.000Z,
              38024,
            ],
            Array [
              2012-04-15T03:00:00.000Z,
              38024,
            ],
            Array [
              2012-04-16T03:00:00.000Z,
              38108,
            ],
            Array [
              2012-04-17T03:00:00.000Z,
              38229,
            ],
            Array [
              2013-02-04T02:00:00.000Z,
              38177,
            ],
            Array [
              2013-02-05T02:00:00.000Z,
              38705,
            ],
            Array [
              2013-02-12T02:00:00.000Z,
              38210,
            ],
            Array [
              2013-02-13T02:00:00.000Z,
              38029,
            ],
            Array [
              2013-02-19T03:00:00.000Z,
              38823,
            ],
            Array [
              2013-02-23T03:00:00.000Z,
              38345,
            ],
            Array [
              2013-02-24T03:00:00.000Z,
              38436,
            ],
            Array [
              2013-03-10T03:00:00.000Z,
              38447,
            ],
          ],
          "diffdata": null,
          "formatters": null,
          "getChartEditor": null,
          "getChartWrapper": [Function],
          "graphID": null,
          "graph_id": null,
          "height": "350px",
          "legendToggle": false,
          "legend_toggle": false,
          "loader": <div>
            Carregando Gráfico
          </div>,
          "options": Object {
            "title": "Red Sox Attendance",
          },
          "render": null,
          "rootProps": Object {
            "data-testid": "1",
          },
          "rows": null,
          "spreadSheetQueryParameters": Object {
            "gid": 1,
            "headers": 1,
          },
          "spreadSheetUrl": null,
          "style": Object {},
          "toolbarID": null,
          "toolbarItems": null,
          "width": "1000px",
        }
      }
    >
      <div>
        Carregando Gráfico
      </div>
      <GoogleChartLoader
        onError={[Function]}
        onLoad={[Function]}
      >
        <Script
          attributes={Object {}}
          onCreate={[Function]}
          onError={[Function]}
          onLoad={[Function]}
          url="https://www.gstatic.com/charts/loader.js"
        />
      </GoogleChartLoader>
    </ContextProvider>
  </Chart>
</CalendarHeatChart>
`;
